safety_prototype/opentelemetry-api/src/opentelemetry/_safety.py

```diff
@@ -0,0 +1,50 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import wraps
+from warnings import warn, resetwarnings
+from traceback import format_exception
+from sys import exc_info
+
+
+def _safe_function(predefined_return_value):
+    """
+    This is the safety mechanism mentioned in the README file.
+
+    This is used as a decorator on every function or method in the API.
+    """
+
+
+    def internal(function):
```

Comment:
Sure, but we can't always return `None`.

Reply:
I guess you're right, but we should change the calling functions to understand that an exception occurred and the returned value is a predefined one... For example, if in your snippet someone calls the safe `function(1,0)` the return value would be `0.0`,  which isn't the true result of `1/0`- it is a result that there was an exception, and the calling function needs to check the return value.
And if we are choosing to do so, we might as well create some object that represents that an exception occurred, and use the option I offered with that value

Reply:
The calling functions do not need to check the return value, they will receive a fake, predefined value when an exception is raised. This is intentional and the way that the user can know that the value is not real is for them to run the application with the `-W error` option so that warnings are turned into exceptions.

Reply:
Ok, makes sense